bin/varnishd/cache/cache_ban_build.c

```diff
@@ -246,7 +246,8 @@ BAN_AddTest(struct ban_proto *bp,
 		return (bp->err);
 
 	for (pv = pvars; pv->name != NULL; pv++)
-		if (!strncmp(a1, pv->name, strlen(pv->name)))
+		if ((!(pv->flag & BANS_FLAG_HTTP) && !strcmp(a1, pv->name))
+			|| (pv->flag & BANS_FLAG_HTTP && !strncmp(a1, pv->name, strlen(pv->name))))
 			break;
```

Comment:
Maybe decompose the loop to make it more readable?

```c
for (pv = pvars; pv->name != NULL; pv++) {
        is_http = pv->flag & BANS_FLAG_HTTP;
        if (!is_http && !strcmp())
            break;
        if (is_http && !strncmp())
            break;
}
```

Otherwise there's probably a line too long, the `||` should be on the previous line and this is the wrong indentation for the if continuation.

Also, this is lacking a test case.